### PR TITLE
Document Spice Runtime ODBC connection configuration

### LINKED ISSUE #318: Add API -> ODBC section

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Document ODBC connection configuration

## 🔨 Related Issues

Closes https://github.com/spiceai/docs/issues/318

image

### CODE REVIEW

File: spiceaidocs/docs/api/odbc/index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+2024-10-06T20:06:50.086948Z  INFO runtime::http: Spice Runtime HTTP listening on 127.0.0.1:8090
+2024-10-06T20:06:50.297512Z  INFO runtime: Initialized results cache; max size: 128.00 MiB, item ttl: 1s
+2024-10-06T20:06:50.308775Z  INFO runtime: Tool [document_similarity] ready to use
+2024-10-06T20:06:50.308803Z  INFO runtime: Tool [table_schema] ready to use
+2024-10-06T20:06:50.308814Z  INFO runtime: Tool [sql] ready to use
+2024-10-06T20:06:50.308829Z  INFO runtime: Tool [list_datasets] ready to use
+2024-10-06T20:06:50.921420Z  INFO runtime: Dataset taxi_trips registered (s3://spiceai-demo-datasets/taxi_trips/2024/), results cache enabled.
+```
+
+Configure client app to use `Arrow Flight SQL ODBC Driver` and run a sample query, such as `SELECT trip_distance, total_amount FROM taxi_trips ORDER BY trip_distance DESC LIMIT 10;`
```
Comment (eadgbear) [inline spiceaidocs/docs/api/odbc/index.md]:
````suggestion
Configure the client app to use `Arrow Flight SQL ODBC Driver` and run a sample query, such as 

```sql
SELECT trip_distance, total_amount FROM taxi_trips ORDER BY trip_distance DESC LIMIT 10;
```

````

File: spiceaidocs/docs/api/odbc/index.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  - [Windows instructions](https://docs.dremio.com/current/sonar/client-applications/drivers/arrow-flight-sql-odbc-driver/#downloading-and-installing-on-windows)
+  - [Linux](https://docs.dremio.com/current/sonar/client-applications/drivers/arrow-flight-sql-odbc-driver/#downloading-and-installing-on-linux)
+  - [macOS instructions](https://docs.dremio.com/current/sonar/client-applications/drivers/arrow-flight-sql-odbc-driver/#downloading-and-installing-on-macos)
+
+### Configure Flight SQL ODBC driver
+
+<Tabs>
+  <TabItem value="default" label="Windows" default>
+    - Open **Start Menu** -> **Windows Administrative Tools** -> click **ODBC Data Sources (64-bit)**
+    - In the **ODBC Data Source Administrator (64-bit)** dialog, click **System DSN**
```
Comment (eadgbear) [inline spiceaidocs/docs/api/odbc/index.md:39]: Can we get a screenshot of the ODBC Data Source Administrator screen?

**Review by eadgbear — APPROVED** [review]

File: spiceaidocs/docs/api/odbc/index.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+2024-10-06T20:06:50.297512Z  INFO runtime: Initialized results cache; max size: 128.00 MiB, item ttl: 1s
+2024-10-06T20:06:50.308775Z  INFO runtime: Tool [document_similarity] ready to use
+2024-10-06T20:06:50.308803Z  INFO runtime: Tool [table_schema] ready to use
+2024-10-06T20:06:50.308814Z  INFO runtime: Tool [sql] ready to use
+2024-10-06T20:06:50.308829Z  INFO runtime: Tool [list_datasets] ready to use
+2024-10-06T20:06:50.921420Z  INFO runtime: Dataset taxi_trips registered (s3://spiceai-demo-datasets/taxi_trips/2024/), results cache enabled.
+```
+
+Configure client app to use `Arrow Flight SQL ODBC Driver` and run a sample query, such as `SELECT trip_distance, total_amount FROM taxi_trips ORDER BY trip_distance DESC LIMIT 10;`
+
```
Comment (eadgbear) [inline spiceaidocs/docs/api/odbc/index.md:118]: Should have an expected output from the example SQL

**Review by eadgbear — DISMISSED** [review]

File: spiceaidocs/docs/api/odbc/index.md (hunk 2 above)
Comment (sgrebnov) [reply spiceaidocs/docs/api/odbc/index.md:39]: Added for Windows and macOS 👍

File: spiceaidocs/docs/api/odbc/index.md (hunk 3 above)
Comment (sgrebnov) [reply spiceaidocs/docs/api/odbc/index.md:118]: I don't think we should show expected output separately as we have it on the screenshot below

**Review by lukekim — DISMISSED** [review]

File: spiceaidocs/docs/api/odbc/index.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
+---
+title: 'ODBC: Open Database Connectivity'
+sidebar_label: 'ODBC'
+sidebar_position: 2
```
Comment (lukekim) [inline spiceaidocs/docs/api/odbc/index.md]:
```suggestion
sidebar_position: 4
```

**Review by eadgbear — APPROVED** [review]

**Pull request merged**